src-backend/api/tests/test_generator.py

```diff
@@ -0,0 +1,274 @@
+
+    def test_element_has_title(self):
+        assert_true('title' in self.attribs)
+        assert_equals(self.attribs['title'], self.procedure.title)
+
+    def test_element_has_author(self):
+        assert_true('author' in self.attribs)
+        assert_equals(self.attribs['author'], self.procedure.author)
+
+    def test_element_has_no_version_attrib(self):
+        assert_false('version' in self.attribs, self.attribs)
+
+    def test_element_has_no_uuid(self):
+        assert_false('uuid' in self.attribs)
+
+    def test_element_has_version_attrib(self):
+        self.procedure.version = '0.1'
+        self.procedure_element = self.generator.generate()
+
+        assert_true('version' in self.procedure_element.attrib)
+        assert_equals(self.procedure_element.attrib['version'], self.procedure.version)
+
+    def test_element_has_uuid_attrib(self):
+        self.procedure.uuid = str(uuid.uuid1())
+        self.procedure_element = self.generator.generate()
+
+        assert_true('uuid' in self.procedure_element.attrib)
+        assert_equals(self.procedure_element.attrib['uuid'], self.procedure.uuid)
+
+
+class PageGeneratorTest(TestCase):
+    def setUp(self):
+        test_user = User.objects.create_user(
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
+        test_user.save()
+
+        procedure = Procedure.objects.create(
+            author='tester',
+            title='test procedure',
+            owner=test_user
+        )
+
+        self.page = Page.objects.create(
+            display_index=0,
+            procedure=procedure
+        )
+
+        self.generator = PageGenerator(self.page)
+        self.page_element = self.generator.generate(ElementTree.Element('test'))
+
+    def test_element_has_correct_name(self):
+        assert_equals(self.page_element.tag, self.generator.name)
+
+    def test_element_has_no_display_index(self):
+        assert_equals(len(self.page_element.attrib), 0)
+
+
+class ElementGeneratorTest(TestCase):
+    def setUp(self):
+        test_user = User.objects.create_user(
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
+        test_user.save()
+
+        procedure = Procedure.objects.create(
+            author='tester',
+            title='test procedure',
+            owner=test_user
+        )
+
+        page = Page.objects.create(
+            display_index=0,
+            procedure=procedure
+        )
+
+        self.element = Element.objects.create(
+            display_index=0,
+            eid='1',
+            element_type='SELECT',
+            concept='HEART SURGERY',
+            question='Which valve',
+            answer='',
+            page=page
+        )
+
+        self.generator = ElementGenerator(self.element)
+        self.elementElement = self.generator.generate(ElementTree.Element('test'))
```

Comment:
they are all element tree elements. It's the naming convention I chose. I can rename it